### PR TITLE
add BulletDebugManager

### PR DESCRIPTION
**Pull request by eundersander**
## Motivation and Context

Debugging tech to help us identify when/why collision-detection is hurting runtime perf.

Sample output from getStepCollisionSummary:
```
[URDF, kitchen_counter, link body, act state 3, group 1, mask -1] vs [RigidObject, ./data/objects/frl_apartment_cup_03.phys_properties.json, id 13, act state 2, group 2, mask -3], 1 points
[URDF, kitchen_counter, link body, act state 3, group 1, mask -1] vs [RigidObject, ./data/objects/frl_apartment_cup_03.phys_properties.json, id 14, act state 2, group 1, mask -1], 4 points
[URDF, fetch, link l_wheel_link, act state 2, group 1, mask -1] vs [Stage, ./orp/start_data/frl_apartment_stage_pvizplan_empty_simplified.stage_config.json, subpart 11, act state 1, group 2, mask -3], 3 points
[URDF, fetch, link r_wheel_link, act state 2, group 1, mask -1] vs [Stage, ./orp/start_data/frl_apartment_stage_pvizplan_empty_simplified.stage_config.json, subpart 11, act state 1, group 2, mask -3], 1 points
[URDF, fetch, link base_link, act state 2, group 1, mask -1] vs [Stage, ./orp/start_data/frl_apartment_stage_pvizplan_empty_simplified.stage_config.json, subpart 11, act state 1, group 2, mask -3], 1 points
```

Primarily, getStepCollisionSummary indicates which objects are colliding on a given step. Act state (activation state), group, and mask are useful extra information if you're diving deep into Bullet collision-detection but can otherwise be ignored.

## How Has This Been Tested

Ad hoc testing of the simulator.

Note that a slightly different version of getPhysicsNumActiveContactPoints has already merged to master https://github.com/facebookresearch/habitat-sim/pull/814 . That version includes a unit test for getPhysicsNumActiveContactPoints. When this change eventually merges to master, the implementation of getPhysicsNumActiveContactPoints here should replace the old one, and the old unit test should be kept unchanged.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/esp/physics/bullet/BulletRigidObject.h [hunk 1]
```diff
@@ -431,6 +431,8 @@ class BulletRigidObject : public BulletBase,
    * updates. See @ref btRigidBody::setWorldTransform. */
   void syncPose() override;
 
+  std::string getBulletDebugName();
```
Comment (jturner65) [inline src/esp/physics/bullet/BulletRigidObject.h:434]: Put in BulletBase.h as pure virtual and override here please.

File: src/esp/physics/bullet/BulletRigidStage.cpp [hunk 2]
```diff
@@ -167,5 +172,10 @@ const Magnum::Range3D BulletRigidStage::getCollisionShapeAabb() const {
   return combinedAABB;
 }  // getCollisionShapeAabb
 
+std::string BulletRigidStage::getBulletDebugName(int subpartId) {
```
Comment (jturner65) [inline src/esp/physics/bullet/BulletRigidStage.cpp:175]: Put in BulletBase.h as pure virtual and override here please.

**Review by jturner65 — COMMENTED** [review]
Looks pretty good.  A small request about the debug string getter :).

**Comment by eundersander** [comment]
The cpp_lint CI fail seems to be in src/esp/io/URDFParser.h, which this PR doesn't touch. I think maybe this branch just doesn't pass our CI right now.

File: src/esp/physics/bullet/BulletRigidStage.cpp (hunk 2 above)
Comment (eundersander) [reply src/esp/physics/bullet/BulletRigidStage.cpp:175]: Why? This signature (including subpartId) is specific to stage.

Comment (jturner65) [reply src/esp/physics/bullet/BulletRigidStage.cpp:175]: If we add other constructs, such as articulated bullet objects, will we want this same debug functionality?

Comment (jturner65) [reply src/esp/physics/bullet/BulletRigidStage.cpp:175]: If so, that's what BulletBase is for, setting up an interface for expected, bullet-specific, functionality we want to see in concretions.

Comment (eundersander) [reply src/esp/physics/bullet/BulletRigidStage.cpp:175]: >If we add other constructs, such as articulated bullet objects, will we want this same debug functionality?
No. subpartId is specific to Stage.

File: src/esp/physics/bullet/BulletRigidObject.h (hunk 1 above)
Comment (eundersander) [reply src/esp/physics/bullet/BulletRigidObject.h:434]: This version of getBulletDebugName (with no arguments) is specific to BulletRigidObject. It is not needed by the other classes that inherit from BulletBase.

File: src/esp/physics/bullet/BulletRigidStage.cpp (hunk 2 above)
Comment (jturner65) [reply src/esp/physics/bullet/BulletRigidStage.cpp:175]: But getBullletDebugName is not unique to stage, is it? The methods for object and stage look nearly identical with the exception of passing an id, which can be defaulted.

Comment (eundersander) [reply src/esp/physics/bullet/BulletRigidStage.cpp:175]: Ok, it looks like we disagree here. You, Alex, and I can discuss on Monday or Tuesday over VC.

Comment (eundersander) [reply src/esp/physics/bullet/BulletRigidStage.cpp]: We chatted offline. I don't think this is appropriate to add to BulletBase. I renamed to getCollisionDebugName to make the method sound less general-purpose.

**Pull request merged**